saleor/graphql/account/mutations/staff/customer_create.py

```diff
@@ -38,3 +52,70 @@ class Meta:
                 description="Setting a new password for the account is requested.",
             ),
         ]
+
+    @classmethod
+    @traced_atomic_transaction()
+    def save(cls, info: ResolveInfo, instance, cleaned_input):
+        manager = get_plugin_manager_promise(info.context).get()
+
+        default_addresses = cls.save_and_apply_addresses_from_input(
+            cleaned_input=cleaned_input, user_instance=instance
+        )
+
+        instance.save()
+
+        if default_addresses["billing"]:
```

Comment:
I am not conviced that current function separation is correct. In `save_and_apply_addresses_from_input` you have  one part of saving process:
- `user_instance.default_shipping_address`

Then as a "post" function `save_and_apply_addresses_from_input` calls, you always have `if`s like:
```python
if default_addresses["billing"]:
    instance.addresses.add(default_addresses["billing"])
```

Calling any of these logic separately, can cause an unexpected results. IMHO  adding to `addresses` & `instance.default_address=` should be wrapped in single function as this is atomic assignment.

Reply:
done